components/menu/stories/template.js

```diff
@@ -396,3 +398,293 @@ export const Template = (
+				{
+					label: "Medium (all purpose)",
+					description: "A medium description about medium is here",
+				},
+				{
+					label: "Large (works best for printing)",
+					description: "A large description about large is here",
+				}
+			],
+		},
+		{
+			heading: "Text truncation with descriptions",
+			shouldTruncate: true,
+			items: [
+				{
+					label: "Small (works best for mobile phones)",
+				},
+				{
+					label: "Medium (all purpose)",
+				},
+				{
+					label: "Large (works best for printing)",
+				}
+			],
+		}
+	];
+
+	return Container({
+		withBorder: false,
+		content: groupData.map((group) => html`
+			${Container({
+				heading: group.heading,
+				content: html`
+					${Template({
+						...args,
+						context,
+						shouldTruncate: group.shouldTruncate || false,
+						items: group.items,
+					})}
+				`
+			})}	
+		`)
+	});
+};
+
+export const SelectionGroup = (args, context) => {
+	const groupData = [
+		{
+			heading: "No selection",
+			items: [
+				{ label: "Cut" },
+				{ label: "Copy" },
+				{ label: "Paste" },
+			],
+		},
+		{
+			heading: "Single selection",
+			selectionMode: "single",
+			items: [
+				{
+					label: "Marquee",
+					isSelected: true,
+				},
+				{
+					label: "Add",
+				},
+				{
+					label: "Subtract",
+				}
+			],
+		},
+		{
+			heading: "Multiple selection with checkboxes",
+			selectionMode: "single",
+			items: [
+				{
+					label: "Marquee",
+					isSelected: true,
+				},
+				{
+					label: "Add",
+				},
+				{
+					label: "Subtract",
+				}
+			],
+		},
+		{
+			heading: "Multiple selection with switches",
+			selectionMode: "multiple",
+			hasActions: true,
+			items: [
+				{
+					label: "Guides",
+					isSelected: true,
+				},
+				{
+					label: "Grid",
+				},
+				{
+					label: "Rulers",
+					isSelected: true,
+				}
+			],
+		},
+	];
+
+	return Container({
+		withBorder: false,
+		content: groupData.map((group) => html`${Container({
+			heading: group.heading,
+			content: html`
+			${Template({
+				...args,
+				context,
+				selectionMode: group.selectionMode || "none",
+				hasActions: group.hasActions || false,
+				items: group.items,
+			})}
+			`
+		})}`)
+	});
+};
+
+export const SubmenuInPopover = (context) => html`${Popover({
+		isOpen: true,
+		position: "end-top",
+		customStyles: {
+			"inline-size": "200px",
+		},
+		trigger: (args, context) => ActionButton({
+			label: "Settings",
+			iconName: "Settings",
+			...args,
+		}, context),
+		content: [
+			(args, context) => Template({
+				items: [
+					{
+						label: "Language",
+						value: "English (US)",
+						isDrillIn: true,
+						isHovered: true,
+					},
+					{
+						label: "Notifications",
+					},
+					{
+						label: "Show grid",
+					}
+				],
+				...args
+			}, context),
+			(args, context) => Popover({
+				isOpen: true,
+				position: "end-top",
+				customStyles: {
+					"--mod-popover-animation-distance": "-4px",
+					top: "-105px",
+					"inline-size": "120px",
```

Comment:
I don't think we handle things like the positioning of the menu next to the submenu in our CSS, so this CSS is making some of those adjustments.